aws/resource_aws_msk_scram_secret.go

```diff
@@ -0,0 +1,216 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/kafka"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceAwsMskScramSecret() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsMskScramSecretCreate,
+		Read:   resourceAwsMskScramSecretRead,
+		Update: resourceAwsMskScramSecretUpdate,
+		Delete: resourceAwsMskScramSecretDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"cluster_arn": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"secret_arn_list": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"scram_secrets": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsMskScramSecretCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).kafkaconn
+
+	existingSecrets, err := readMSKClusterSecrets(conn, d.Get("cluster_arn").(string))
+	if err != nil {
+		return fmt.Errorf("failed lookup secrets %s", err)
+	}
+
+	createSecrets := filterMskNewSecrets(expandStringList(d.Get("secret_arn_list").([]interface{})), existingSecrets)
```

Comment:
the function signature for `filterMskNewSecrets` shows the `existingSecrets` following by `newSecrets` but here we have what seems to be `newSecrets` following by `existingSecrets`, should these be reversed?

Reply:
the goal of this function is to determine which secret in `secret_arn_list` is not in the `existingSecrets` list determined from the call to `readMSKClusterSecrets` above?

Reply:
noting here if we determine `secret_arn_list` can in fact be represented as a Set instead this filtering can be updated to use the Set functions as:
```
existingSecretsSet := schema.NewSet(schema.HashString, existingSecrets)
newSecrets := d.Get("secret_arns").(*schema.Set)
createSecrets := newSecrets.Difference(existingSecretsSet)
```